fanficfare/adapters/adapter_spiritfanfictioncom.py

```diff
@@ -0,0 +1,350 @@
+            else:
+                i += 1
+
+        # Classificação, Gêneros e Avisos
+        # Finding div element with class "clearfix baixo"
+        div_element = soup.find('div', {'class': 'clearfix baixo'})
+
+        # Finding div element with class "classificacao"
+        classificacao_element = div_element.find('div', class_='classificacao')
+
+        # Extracting last word from class name
+        if classificacao_element and 'class' in classificacao_element.attrs:
+            class_value = classificacao_element.attrs['class']
+            rating = class_value[-1].split('-')[-1]
+
+        # Extracting text content "Gêneros" and "Avisos"
+        contents = classificacao_element.find_next('div').contents
+        for i, content in enumerate(contents):
+            if content.text.strip() == 'Gêneros:':
+                parse_until_br('genre', i+1, contents)
+            if content.text.strip() == 'Avisos:':
+                parse_until_br('warnings', i+1, contents)
+
+    ## Normalize chapter URLs in case of title change
+    def normalize_chapterurl(self,url):
+        #https://www.spiritfanfiction.com/historia/story-name-1234/capitulo56
+        url = re.sub(r"https?://("+self.getSiteDomain()+r"/historia/\d+/capitulo\d+)$",
+                     r"https://\1",url)
+        return url
+
+
+    def getChapterText(self, url):
+        logger.debug('Getting chapter text from: %s' % url)
+
+        save_chapter_soup = self.make_soup(self.get_request(url))
+        ## need save_chapter_soup for .new_tag()
+        save_chapter = BeautifulSoup("", "html.parser").new_tag("div")
+
+        chapter_dl_soup = self.make_soup(self.get_request(url))
+        if None == chapter_dl_soup:
+            raise exceptions.FailedToDownload("Error downloading Chapter: %s!  Missing required element!" % url)
+        div_chapter = chapter_dl_soup.find('div', {'class':'clearfix'})
+        chapter_text = div_chapter.find('div', class_='texto-capitulo')
+
+        exclude_notes=self.getConfigList('exclude_notes')
+
+        def decode_email(encoded_email):
+            email = ""
+            r = int(encoded_email[:2], 16)
+            for i in range(2, len(encoded_email), 2):
+                char_code = int(encoded_email[i:i + 2], 16) ^ r
+                email += chr(char_code)
+            return email
+        
+
+        def decode_emails(html_text):
+            # Find all elements with class '__cf_email__'
+            email_elements = html_text.find_all(class_='__cf_email__')
+            for element in email_elements:
+                # Get the data-cfemail attribute value
+                encoded_email = element.get('data-cfemail')
+                if encoded_email:
+                    # Decode the email address
+                    decoded_email = decode_email(encoded_email)
+                    # Replace the obfuscated email with the decoded email
+                    element.string = decoded_email
+            return str(html_text)
```

Comment:
`str()` calls all need to be `unicode()` for backward compatibility for python2 in older versions of Calibre.